tests/Test.Yaapii.Xambly/Directive/NsDirectiveTest.cs

```diff
@@ -48,40 +49,262 @@ public void SetsNsAttr()
         {
-            //var root = new XElement("html",
-            //                new XElement("head")
-            //                //new XElement("body")
-            //            );
             var dom = new XDocument();
+            new Xambler(
+                new Directives()
+                .Add("root")
+                    .Add("parent")
+                        .Add("child")
+                        .Set("child value")
+                .Xpath("/root")
+                .Ns("MyDefaultNamespaceUri")
+            ).Apply(dom);
 
+            Assert.Equal(
+                "<root xmlns=\"MyDefaultNamespaceUri\"><parent><child>child value</child></parent></root>",
+                dom.ToString(SaveOptions.DisableFormatting)
+            );
+        }
+
+        [Fact]
+        public void ReplacesDefaultNamespace()
+        {
+            var dom = new XDocument();
+            new Xambler(
+                new Directives()
+                .Add("root")
+                .Ns("OldValue")
+                .Ns("MyDefaultNamespaceUri")
+            ).Apply(dom);
+
+            Assert.Equal(
+                "<root xmlns=\"MyDefaultNamespaceUri\" />",
+                dom.ToString(SaveOptions.DisableFormatting)
+            );
+        }
+
+        [Fact]
+        public void AddsDefaultNamespaceToSelectedNode()
+        {
+            var dom = new XDocument();
+            new Xambler(
+                new Directives()
+                .Add("root")
+                    .Add("parent")
+                        .Add("child")
+                .Xpath("/root/parent")
+                .Ns("MyDefaultNamespaceUri")
+            ).Apply(dom);
+
+            Assert.Equal(
+                "<root><parent xmlns=\"MyDefaultNamespaceUri\"><child /></parent></root>",
+                dom.ToString(SaveOptions.DisableFormatting)
+            );
+        }
+
+        [Fact]
+        public void AddsDefaultNamespaceToSeveralNodes()
+        {
+            var dom = new XDocument();
             new Xambler(
-                new AddDirective("html"),
-                new NsDirective("http://www.w3.org/1999/xhtml"),
-                new AddDirective("head"),
-                new UpDirective(),
-                new AddDirective("body")
+                new Directives()
+                .Add("root")
+                    .Add("parent")
+                        .Add("child")
+                .Xpath("/root")
+                    .Add("anotherParent")
+                        .Add("anotherChild")
+                .Xpath("/root/*")
+                .Ns("MyDefaultNamespaceUri")
             ).Apply(dom);
 
+            Assert.Equal(
+                "<root><parent xmlns=\"MyDefaultNamespaceUri\"><child /></parent><anotherParent xmlns=\"MyDefaultNamespaceUri\"><anotherChild /></anotherParent></root>",
+                dom.ToString(SaveOptions.DisableFormatting)
+            );
+        }
+
+        [Fact]
+        public void DefaultNamespaceAffectesXPath()
```

Comment:
How does this test that xpath regards the namespace?

Reply:
See line 155 `.Xpath("/def:root/def:parent")`
If the xpath is affected here

Reply:
You mean the term "affects" - ok
Changed to `DefaultNamespaceIsRequiredInXPath`